components/public-api/gitpod/experimental/v1/workspaces.proto

```diff
@@ -86,6 +91,16 @@ message StopWorkspaceRequest {
 }
 message StopWorkspaceResponse {}
 
+message SendHeartbeatRequest {
+    string workspace_id = 1;
+}
```

Comment:
If we extend this struct with, we can avoid having an RPC for every property in our domain model
```suggestion
message SendHeartbeatRequest {
    string workspace_id = 1;
    // indicates this is the terminal heartbeat and no further ones will be sent. This happens when the IDE closes.
    bool terminal = 2;
}
```

Reply:
Could you elaborate which property?

We would like to split 2 different functions `SendHearbeat` and report that a client closed. Before it was mixed in one but turned out to be orthogonal things from client perspective.

Reply:
If they are orthogonal, then that's fine and should be seperate RPCs. From the implementation it did look like they are coupled but I don't know your domain that well so that makes sense.